pulp_ansible/app/viewsets.py

```diff
@@ -70,6 +73,38 @@ class CollectionFilter(ContentFilter):
     FilterSet for Ansible Collections.
     """
 
+    latest = BooleanFilter(field_name='latest', method='filter_latest')
+
+    def filter_latest(self, queryset, name, value):
+        """
+        If the value of 'latest' is True, exclude all non-latest Collections from the results.
+
+        Args:
+            queryset: The already-formed queryset for modification
+            name: The name of the parameter, 'latest'
+            value: The value of the argument. This is checked if 'True' or not.
+
+        Returns:
+            Queryset with non-latest collections excluded if value is True.
+        """
+        if value:
+            namespace_name_dict = defaultdict(lambda: defaultdict(list))
+            for collection in queryset.all():
+                version_entry = (parse(collection.version), collection.pk)
+                namespace_name_dict[collection.namespace][collection.name].append(version_entry)
+
+            not_latest_pks = []
+            for namespace, name_dict in namespace_name_dict.items():
+                for name, version_list in name_dict.items():
```

Comment:
Does python provide a best practice for naming unused variables?

Reply:
I don't think it does, but it strives for readability as a first-class concern. There is some talk of including variable typing as an anti-pattern, but I think the jury is still out on that. Also having the types to me makes them more comprehensible.

Reply:
I'm not sure I understand. What does typing have to do with naming?